### DISCUSSION TITLE
Issue with white Logo on transparent background

### DISCUSSION THREAD

**Discussion by mirko-horvacki** [dlemstra/Magick.NET#1326] [Help] [answered]
Hi,

I tried to create PNG files from 2 files which can be downloaded from following URL:
[https://easyupload.io/m/wz6k6m](https://easyupload.io/m/wz6k6m)

Result is not same as in Photoshop - in Photoshop images are white logos on transparent background, but PNG`s created with usage of MagickImage contained only one color - on one case only white, and in second - only transparent.

Here is code snippet used to create PNGs:

`MagickReadSettings settingsVectorCheckSize = new MagickReadSettings();`
`settingsVectorCheckSize.AntiAlias = false;`
`settingsVectorCheckSize.Density = new Density(imageDPi, DensityUnit.PixelsPerInch);`
`settingsVectorCheckSize.ColorSpace = ColorSpace.sRGB;`
`MagickImage imageCheckSize = new MagickImage(byteArray, settingsVectorCheckSize);`
`imageCheckSize.Format = MagickFormat.Png;`

Is it possible to rasterize used AI and PDF to PNG with two colors - white and transparent?

Thanks in advance!

Kind regards,

Mirko

**Comment by dlemstra** [comment]
Adding images here instead: [images.zip](https://github.com/dlemstra/Magick.NET/files/10359125/images.zip)

**Comment by dlemstra** [comment, marked as answer]
This library uses Ghostscript and it seems that it decides to use a white background. We are using the following options:

`-q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 "-sDEVICE=pngalpha" -dTextAlphaBits=4 -dGraphicsAlphaBits=4 "-r72x72" -dPrinted=false  "-sOutputFile=out.png" 1.ai`

This is executed through a library or on the command line with `gswin64c.exe`. If you figure out which option should be added to change the background then I might be able to integrate that into ImageMagick.

**Reply by mirko-horvacki** [reply]
Asked question on GhostScript site https://bugs.ghostscript.com/show_bug.cgi?id=706294, got answer - files can not be processed correctly with GhostScript.

**Reply by dlemstra** [reply]
Thanks for the followup and coming back to report your findings
